src/buffer.cc

```diff
@@ -204,6 +205,82 @@ Result Buffer::CompareRMSE(Buffer* buffer, float tolerance) const {
   return {};
 }
 
+std::vector<double> Buffer::GetHistogramForChannel(uint32_t channel,
+                                                   uint32_t num_bins) const {
+  std::vector<double> bins(num_bins, 0.0);
+  auto* buf_ptr = GetValues<uint8_t>();
+  const auto& segments = format_->GetSegments();
+  const auto comp = segments[channel].GetComponent();
+  auto num_channels = format_->ValuesPerElement();
+  buf_ptr += channel * comp->SizeInBytes();
+  for (size_t i = 0; i < ElementCount(); ++i) {
+    const auto bin = *reinterpret_cast<const uint8_t*>(buf_ptr);
+    bins[bin]++;
+    // Assumes unpacked format where all channels have the same bit size
+    buf_ptr += comp->SizeInBytes() * num_channels;
+  }
+
+  return bins;
+}
+
+Result Buffer::CompareHistogramEMD(Buffer* buffer, float tolerance) const {
```

Comment:
Is it possible to add unit tests for this?

Reply:
You mean under parser_expect_test.cc?

Reply:
in buffer_test.cc

Reply:
Ok, I'll ask another team member to write me those. I will push the other changes before that.

Reply:
Btw. the RMSE tests are under parser_expect_test.cc. Should we have all in the same place for consistency?

Reply:
I'm not sure if I added unit tests for RMSE or just parser tests. I think I didn't add unit tests at the time I wrote it, so we should add some in the buffer class at some point in the future.

Reply:
Should we do separate issues for both so this PR won't have to wait for the unit tests?